crates/rethnet_eth/src/remote/client.rs

```diff
@@ -366,16 +371,53 @@ impl RpcClient {
         cache_key: &str,
         result: impl Serialize,
     ) -> Result<(), RpcClientError> {
-        let cache_path = self.make_cache_path(cache_key).await?;
         let contents = serde_json::to_string(&result).expect(
             "result serializes successfully as it was just deserialized from a JSON string",
         );
-        match tokio::fs::write(cache_path, contents).await {
+
+        // 1. Write to a random temporary file first to avoid race conditions.
+        let tempfile = match tokio::task::spawn_blocking(NamedTempFile::new).await? {
```

Comment:
Is the overhead of spawning a task worth it for the creation of the file? This feels like premature optimisation.

Reply:
It's required unfortunately, otherwise it'd be blocking the worker threads as `NamedTempFile::new` does blocking i/o. 

(The `tokio::fs` module does the same under the hood to wrap blocking `std::fs` operations.)